src/esp/assets/ResourceManager.cpp

```diff
@@ -991,8 +991,10 @@ bool ResourceManager::loadGeneralMeshData(
 
     // if this is a new file, load it and add it to the dictionary
     LoadedAssetData loadedAssetData{info};
-    loadTextures(*fileImporter_, loadedAssetData);
-    loadMaterials(*fileImporter_, loadedAssetData);
+    if (requiresTextures_) {
+      loadTextures(*fileImporter_, loadedAssetData);
+      loadMaterials(*fileImporter_, loadedAssetData);
+    }
```

Comment:
Just a note that `Simulator.reconfigure` won't reload these after skipping them if the configuration changes without first calling `close()`.